### DISCUSSION TITLE
Spellchecking on Windows

### DISCUSSION THREAD

**Discussion by K4Werri** [zim-desktop-wiki/zim-desktop-wiki#2047] [Q&A] [answered]
Hi,

I'm using v0.73.2 and notice that the spell checking is unavailable on Windows.

Is this the case or have I made a mistake with the installation?

Also, I found a page, https://github.com/jaap-karssenberg/zim-wiki/wiki/Install-spell-checking-on-windows, that describes a process for installing the spell checking function on WIndows but note that this advice was edited in 2014.

Is this still the current advice for getting spell checking on WIndows?

Thanks.

Kevin.

**Comment by micha06de** [comment]
The instructions from 2014 will no longer work, since they still refer to Python 2.x.  
Why is the topic of spell checking under Windows treated so stepmotherly here?
Unfortunately, I haven't found a way to make the spell checker work under Windows either.

**Comment by cantosciaran** [comment]
For Windows installer - Zim Desktop 0.75.2

I Couldn't get pygtkspellcheck installed into Python3.12 via pip due to dependence on PyGObject,  and (if I understand correctly) the only supported method of compiling being via Msys2/mingw which is well outside my comfort/knowledge windows.

But I did get gtkspell working.  Here's a summary of what I did.

Install Python (used 3.11.7 - v3.12 won't work)
Install pygi-aio-3.24.1-rev1-setup (https://sourceforge.net/projects/pygobjectwin32/files/)
- default path -> should be the \python311\lib\site-packages folder of your install -> next
- Packages:
     * GNOME/Freedesktop libraries:  base packages -> Next
     * Non-GNOME libraries:  GTKSpell 3.0.3 -> Next
     * Dev packages:  none -> Next
     * confirm &tc. -> Next
Open CMD - go to your Python3.11\scripts folder and install pyenchant:
pip install pyenchant 

(pygi-aio will give you the spellcheck support.  pyenchant will give you the dictionaries.)


I install my Zim as portable using the Windows installer.

You want to copy the following to your Zim install folder:

Dictionaries:
From:  \Python311\Lib\site-packages\enchant\data\mingw64\share\enchant\hunspell
To:  \Zim\share\enchant\myspell   (likely need to create the folder)

Spellcheck support:
    files:  libenchant-1.dll, libgtkspell3-3-0.dll, libstdc++.dll
        From:  \Python311\Lib\site-packages\gnome
        to:  \Zim\   (root of zim install)
    folders:  enchant, girepository-1.0
        From: \Python311\Lib\site-packages\gnome\lib
        To:  \Zim\lib

Launch Zim -> Edit -> Preferences -> plugins -> spell checker
     Check the box to enable
     -> Properties:  Set your dictionary language (whatever code matches your language in the dictionaries you copied) , or edit your environ.ini to set your # Interface language

Tools -> Check spelling (or F7) to enable/disable.

You can uninstall pygi-aio and python if you don't need them for anything else.

**Reply by introt** [reply]
Nice work! To clarify, msys2/mingw is for building the installer, Zim is written in Python and is never compiled. What you've demonstrated here is that it is possible to install stuff on top of Zim's portable installation, so pygtkspellcheck should also work - you just need the correct environment variables (PATH and PYTHONPATH, presumably the root of the Zim install and Zim\lib, respectively) for pip to detect the dependencies. You could probably skip installing another Python and copying stuff over in a similar way.

**Reply by tridian-tn** [reply]
FWIW I've just followed these instructions and they work perfectly. I used:

* a later point release of Python 3.11 (3.11.8)
* the pygi-aio-3.24.1-rev1-setup installer, as described
* the non-portable install of Zim, where the base directory is `C:\Program Files\Zim Desktop Wiki`

**Reply by tridian-tn** [reply, marked as answer]
Just to get this gtkspell support documented, I've updated https://github.com/jaap-karssenberg/zim-wiki/wiki/Install-spell-checking-on-windows - please feel free to revise!

**Reply by eduardomozart** [reply]
I was able to get it working on Python 3.11.9 and latest Zim release (0.76.3) following the original instructions.
Enchant comes only with ``en_*`` dictionaries.
If you are using a foreign language, you can download the ``*.dic`` and ``.aff`` files from [LibreOffice](https://github.com/LibreOffice/dictionaries/tree/master) for your language ([source](https://stackoverflow.com/questions/67678516/pyenchant-for-other-languages)) and copy them to the ``myspell`` folder if Zim fails to load the dictionary for your language.
To spell checking to work, also make sure to check the "Spell checking" box into "Tools" menu after enabling the plugin.

**Reply by tridian-tn** [reply]
@eduardomozart I updated the support doc above with what you found. Happy for you to amend if you'd prefer it worded differently.
